libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_parser.c

```diff
@@ -285,6 +341,50 @@ static void _testCreateKeysAndCertificateAcceptedCallback( void * contextParam,
 
 /*-----------------------------------------------------------*/
 
+static void _testCreateCertFromCsrRejectedCallback( void * contextParam,
+                                                    const AwsIotProvisioningCreateCertFromCsrResponse_t * pResponseInfo )
+{
+    AwsIotProvisioningRejectedResponse_t * pExpectedParams =
+        ( AwsIotProvisioningRejectedResponse_t * ) contextParam;
+
+    AwsIotProvisioning_Assert( pResponseInfo->statusCode == _expectedStatusCode );
+    /* Verify that the rejected response was parsed as expected. */
+    _verifyParsedRejectedResponse( pExpectedParams, &pResponseInfo->u.rejectedResponse );
+}
+
+/*-----------------------------------------------------------*/
+
+static void _testCreateCertFromCsrAcceptedCallback( void * contextParam,
+                                                    const AwsIotProvisioningCreateCertFromCsrResponse_t * pResponseInfo )
+{
+    AwsIotProvisioningCreateCertFromCsrResponse_t * pExpectedParams =
+        ( AwsIotProvisioningCreateCertFromCsrResponse_t * ) contextParam;
+
+    /* Disable unused variable warnings. */
+    ( void ) pExpectedParams;
+    ( void ) pResponseInfo;
+
+    /* Verify that the rejected response was parsed as expected. */
+    AwsIotProvisioning_Assert( pResponseInfo->statusCode == AWS_IOT_PROVISIONING_SERVER_STATUS_ACCEPTED );
+
+    AwsIotProvisioning_Assert(
+        pExpectedParams->u.acceptedResponse.deviceCertLength ==
+        pResponseInfo->u.acceptedResponse.deviceCertLength );
+    AwsIotProvisioning_Assert( pExpectedParams->u.acceptedResponse.pDeviceCertificate ==
+                               pResponseInfo->u.acceptedResponse.pDeviceCertificate );
+    AwsIotProvisioning_Assert(
+        pExpectedParams->u.acceptedResponse.certIdLength ==
+        pResponseInfo->u.acceptedResponse.certIdLength );
+    AwsIotProvisioning_Assert( pExpectedParams->u.acceptedResponse.pCertId ==
```

Comment:
This file uses ```AwsIotProvisioning_Assert``` and Unity Asserts ```TEST_ASSERT_EQUAL```. Why not use Unity Asserts at all the places?

Reply:
I had earlier noticed that the unity asserts don't work outside of the test group/case constructs..
I can verify again to be sure.

Reply:
Verified that it works with the Unity test assert (contrary to my last year's observation). Have updated to use unity test macros